presto-spi/src/main/java/com/facebook/presto/spi/type/CharType.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.util.Objects;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
+
+public final class CharType
+        extends AbstractVariableWidthType
+{
+    public static final int MAX_LENGTH = 65_536;
+
+    public static CharType createCharType(long length)
+    {
+        return new CharType(length);
+    }
+
+    private final int length;
+
+    private CharType(long length)
+    {
+        super(
+                new TypeSignature(
+                        StandardTypes.CHAR,
+                        singletonList(TypeSignatureParameter.of(length))),
+                Slice.class);
+
+        if (length < 1 || length > MAX_LENGTH) {
```

Comment:
SQL standard specifies that minimum CHAR length is 1. See: https://mariadb.com/kb/en/sql-99/character-string-data-types/